src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java

```diff
@@ -75,22 +76,21 @@ public void importEntries(List<BibEntry> entriesToImport) {
         // Check if we are supposed to warn about duplicates.
         // If so, then see if there are duplicates, and warn if yes.
         if (preferences.shouldWarnAboutDuplicatesForImport()) {
-            boolean containsDuplicate = entriesToImport.stream()
-                                                       .anyMatch(this::hasDuplicate);
-
-            if (containsDuplicate) {
-                boolean continueImport = dialogService.showConfirmationDialogWithOptOutAndWait(Localization.lang("Duplicates found"),
-                        Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
-                        Localization.lang("Continue with import"),
-                        Localization.lang("Cancel import"),
-                        Localization.lang("Disable this confirmation dialog"),
-                        optOut -> preferences.setShouldWarnAboutDuplicatesForImport(!optOut));
-
-                if (!continueImport) {
-                    dialogService.notify(Localization.lang("Import canceled"));
-                    return;
+            BackgroundTask.wrap(() -> entriesToImport.stream()
+                    .anyMatch(this::hasDuplicate)).onSuccess(e -> {
+                if (e) {
+                    boolean continueImport = dialogService.showConfirmationDialogWithOptOutAndWait(Localization.lang("Duplicates found"),
+                            Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
+                            Localization.lang("Continue with import"),
+                            Localization.lang("Cancel import"),
+                            Localization.lang("Disable this confirmation dialog"),
+                            optOut -> preferences.setShouldWarnAboutDuplicatesForImport(!optOut));
+
+                    if (!continueImport) {
+                        dialogService.notify(Localization.lang("Import canceled"));
```

Comment:
There is a small problem with the current solution: because the background task is called asynchronously, the `importEntries` is always called even if there are duplicates. That is the code now runs in the following order: import entries > duplicate check > ask user. But the desired order is duplicate check > ask user > import entries.

Reply:
The importEntries depends on duplicateCheck from the beginning it seems.

Does it mean the importEntries requires some refactoring so that duplicate check will not be called inside that method?

Reply:
The importEntries depends on duplicateCheck from the beginning it seems.

Does it mean the importEntries requires some refactoring so that duplicate check will not be called inside that method?

Reply:
I think it should be enough to move the rest of the method up here:
````
if (!continueImport) {
    dialog.notify(aborted);
} else {
    init file handler
    import();
}
````

Reply:
Thanks for the update. It looks almost good now: the import should also be performed `if (preferences.shouldWarnAboutDuplicatesForImport())` returns false and if there is no duplicates (i.e `e = false`).

Reply:
thanks, sure

Reply:
Same as abvoe, please no abbreviations like "e" for variables (except for exceptions maybe)